app/controller/AppController.php

```diff
@@ -65,25 +65,55 @@ public function accumulate(string $value)
     {
         try {
             $this->manager->append($value);
-            $chaine = $this->manager->getAccumulator();
-            $this->index($chaine);
-        } catch(Exception $e) {
+            $this->index();
+        } catch (Exception $e) {
             $this->error($e->getMessage());
         }
     }
 
     /**
      * Lance une action (opération, égal, pourcentage, ...)
      * @param string $action
+     * @throws Exception
      */
     public function action(string $action)
     {
-        switch($action) {
-            case 'clear';
+        $result = '';
+        $input = '';
+        switch ($action) {
+            case 'clear':
                 $this->manager->reset();
+                $result = $this->manager->getResult();
+                break;
+            case 'plus':
+                $this->manager->operator(Calculator::PLUS);
+                $result = $this->manager->getResult();
+                $input = $this->manager->getInput();
+                break;
```

Comment:
globalement dans le cas des opérations à 2 opérandes (division, multiplication, soustraction, addition) tu fais exactement le même code, la seule chose qui change c'est l’opérateur que tu passes en paramètres à la méthode manager->operator().

Tu peux donc imaginer créer une méthode operation qui prendrait un paramètre et qui exécuterais le code que tu as entre chaque `case`.

d'autre part pour éviter d'avoir dans chacun de ces cases le même appel à cette médode operation, tu peux écrire les choses comme ça :

`case 'divide':
  case 'times':
  case 'minus':
  case 'plus':
      $this->manager->operation(...);
      break;
...
`
C'est juste un exemple. A toi de voir comment gérer les arguments etc.